internal/database/repository/queue.go

```diff
@@ -3,176 +3,65 @@ package repository
-		preempting_resource,
-		head_room,
-		is_leaf,
-		is_managed,
-		properties,
-		parent,
-		template_info,
-		abs_used_capacity,
-		max_running_apps,
-		running_apps,
-		current_priority,
-		allocating_accepted_apps,
-    	created_at) VALUES (@id, @parent_id, @queue_name, @status, @partition, @pending_resource, @max_resource,
-		@guaranteed_resource, @allocated_resource, @preempting_resource, @head_room, @is_leaf, @is_managed, @properties,
-		@parent, @template_info, @abs_used_capacity, @max_running_apps, @running_apps,
-		@current_priority, @allocating_accepted_apps, @created_at)
-	ON CONFLICT (partition, queue_name) DO UPDATE SET
-		status = EXCLUDED.status,
-		pending_resource = EXCLUDED.pending_resource,
-		max_resource = EXCLUDED.max_resource,
-		guaranteed_resource = EXCLUDED.guaranteed_resource,
-		allocated_resource = EXCLUDED.allocated_resource,
-		preempting_resource = EXCLUDED.preempting_resource,
-		head_room = EXCLUDED.head_room,
-		is_leaf = EXCLUDED.is_leaf,
-		is_managed = EXCLUDED.is_managed,
-		max_running_apps = EXCLUDED.max_running_apps,
-		running_apps = EXCLUDED.running_apps`
-	for _, q := range queues {
-		parentId, err := s.getQueueID(ctx, q.Parent, q.Partition)
-		if err != nil {
-			return fmt.Errorf("could not get parent queue from DB: %v", err)
-		}
-		_, err = s.dbpool.Exec(ctx, upsertSQL,
-			pgx.NamedArgs{
-				"id":                       ulid.Make().String(),
-				"parent_id":                parentId,
-				"queue_name":               q.QueueName,
-				"status":                   q.Status,
-				"partition":                q.Partition,
-				"pending_resource":         q.PendingResource,
-				"max_resource":             q.MaxResource,
-				"guaranteed_resource":      q.GuaranteedResource,
-				"allocated_resource":       q.AllocatedResource,
-				"preempting_resource":      q.PreemptingResource,
-				"head_room":                q.HeadRoom,
-				"is_leaf":                  q.IsLeaf,
-				"is_managed":               q.IsManaged,
-				"properties":               q.Properties,
-				"parent":                   q.Parent,
-				"template_info":            q.TemplateInfo,
-				"abs_used_capacity":        q.AbsUsedCapacity,
-				"max_running_apps":         q.MaxRunningApps,
-				"running_apps":             q.RunningApps,
-				"current_priority":         q.CurrentPriority,
-				"allocating_accepted_apps": q.AllocatingAcceptedApps,
-				"created_at":               time.Now().Unix(),
-			})
-		if err != nil {
-			return fmt.Errorf("could not insert/update queue into DB: %v", err)
-		}
-	}
-	return nil
-}
-
-func (s *PostgresRepository) AddQueues(ctx context.Context, parentId *string, queues []*dao.PartitionQueueDAOInfo) error {
+func (s *PostgresRepository) InsertQueue(ctx context.Context, q *model.Queue) error {
 	insertSQL := `INSERT INTO queues (
-		parent_id, queue_name, status, partition, pending_resource, max_resource,
+		id, created_at_nano, queue_name, parent_id, status, partition, pending_resource, max_resource,
```

Comment:
I think this `parent_id` is not inserted properly. `buildPartitionQueueTrees` method in the webservice depends on this. Now `buildPartitionQueueTrees` might give a flat list rather than a tree.

Reply:
We need @richscott 's changes to set the id so we can reliably set these parameters

Reply:
I am literally adding that to the `Queue` type and funcs right now, then just a few other types, in yk core. It will arrive very soon.